Steepfile

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+target :lib do
+  signature "sig"
+
+  # Tell Steep about stdlib require's
+  library "ripper" # RipperCompat
+  library "delegate"  # LexCompat
+  library "cgi"  # DotVisitor
```

Comment:
Re: Ripper usage

* None of the public RBS files mention Ripper anymore
* `rake check_annotations` validates this because the `-r prism` call ignores dirs starting with `_` by convention.
* The gemspec only includes the public RBS files and no longer includes a manifest.yml

Now Steep is used within the project to statically type-check the implementations against the RBS files. The comment-based type annotations are still limited, so I've added RBS declarations as needed in the sig/prism/_private folder.

* Since there are `require`s to ripper, delegate, and cgi in various files in lib/prism/*.rb, I need to tell Steep about them so it can load the stdlib RBS files that are inside the `rbs` gem. Core Ruby RBS is loaded automatically.
* My hunch is steep doesn't automatically pull them in when it sees the `require`s because LOAD_PATH is dynamic, so it can't statically work that out.